analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.cs

```diff
@@ -172,5 +172,83 @@ public void Execute_PersistSymbols_InsideBlock()
             static ISymbol LocalReferenceOperationSymbol(IOperationWrapperSonar operation) =>
                 ((ILocalReferenceOperation)operation.Instance).Local;
         }
+
+        [TestMethod]
+        public void ExitReached_SimpleFlow()
+        {
+            var exitCount = 0;
+            SETestContext.CreateCS("var a = true;", new ExitReachedTestCheck(state => CountExits(state, ref exitCount)));
```

Comment:
This thing is used 5 times `new ExitReachedTestCheck(state => CountExits(state, ref exitCount)));` exactly the same way and it's complicated (double indirection, class, method, lambda, lot of copy/paste keystrokes).

Let's add this functionality to `CollectorTestCheck` 
```
        public override ProgramState ExitReached(ProgramState state) =>
            exitReachedCount++;
```

and assert it like this:
```
public void ExitReached_SimpleFlow() =>
    SETestContext.CreateCS("var a = true;").Collector.ValidateExitReachCount(1);
```

Reply:
I think it's a nice idea which will simplify things. What do you think about changing the collector to just gather data and letting the test to do the assertion?

Reply:
I'd keep the pattern of having assertions inside the collector. We will soon need to collect and assert more stuff there (collect states, assert on them) and it will be easier to modify the internals of this single class then updating every single usage.
It also makes the code in UTs a bit shorter.

Reply:
We will have a mix of responsibilities there, collection + validation. In my opinion, we should separate those.

Reply:
As discussed offline, I've extracted the validate method and in the next PR I will rename the collector to validator.